projects/igniteui-angular/src/lib/paginator/paginator.component.ts

```diff
@@ -0,0 +1,247 @@
+import { CommonModule } from '@angular/common';
+import { FormsModule } from '@angular/forms';
+import { Component, Input, Output, NgModule, Optional, Inject, EventEmitter, HostBinding } from '@angular/core';
+import { CurrentResourceStrings } from '../core/i18n/resources';
+import { IDisplayDensityOptions, DisplayDensityToken, DisplayDensityBase, DisplayDensity } from '../core/displayDensity';
+import { IgxSelectModule } from '../select/index';
+import { IgxIconModule } from '../icon/index';
+import { IgxButtonModule } from '../directives/button/button.directive';
+import { IgxRippleModule } from '../directives/ripple/ripple.directive';
+
+@Component({
+    selector: 'igx-paginator',
+    templateUrl: 'paginator.component.html',
+})
+export class IgxPaginatorComponent extends DisplayDensityBase {
+
+    /**
+     * Total pages calculated from totalRecords and perPage
+     */
+    public totalPages: number;
+    protected _page = 0;
+    protected _totalRecords: number;
+    protected _selectOptions = [5, 10, 15, 25, 50, 100, 500];
+    protected _perPage = 15;
+
+    /**
+     * @hidden
+     */
+    @HostBinding('class')
+    public get classes(): string {
+        switch (this.displayDensity) {
+            case DisplayDensity.cosy:
+                return 'igx-grid-paginator--cosy';
+            case DisplayDensity.compact:
+                return 'igx-grid-paginator--compact';
+            default:
+                return 'igx-grid-paginator';
+        }
+    }
+
+    /**
+     * @hidden
+     */
+    @Input()
+    public get page() {
+        return this._page;
+    }
+
+    public set page(value: number) {
+        this._page = value;
+        this.pageChange.emit(this._page);
+    }
+
+    /**
+   * An @Input property, sets number of visible items per page in the `IgxPaginatorComponent`.
+   * The default is 15.
+   * ```typescript
+   * let itemsPerPage = this.paginator.perPage;
+   * ```
+   * @memberof IgxPaginatorComponent
+   */
+    @Input()
+    public get perPage() {
+        return this._perPage;
+    }
+
+    public set perPage(value: number) {
+        this._perPage = value;
```

Comment:
IMO changing the `perPage` property should call the `sortUniqueOptions` method, otherwise custom values
are not shown in the select drop-down.